pkg/ansible/watches/watches.go

```diff
@@ -136,10 +136,53 @@ func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	w.WatchClusterScopedResources = tmp.WatchClusterScopedResources
 	w.Finalizer = tmp.Finalizer
 	w.AnsibleVerbosity = getAnsibleVerbosity(gvk, ansibleVerbosityDefault)
+	w.addRolePlaybookPaths()
 
 	return nil
 }
 
+// addRolePlaybookPaths will add the full path for rules an playbooks
+// when the relative one is informed. Note that these files are moved in the build
+// to /opt/ansible always
+func (w *Watch) addRolePlaybookPaths() {
+	w.Playbook = getFullPath(w.Playbook)
+	w.Role = getFullPath(w.Role)
+	if w.Finalizer != nil && len(w.Finalizer.Role) > 0 {
+		w.Finalizer.Role = getFullPath(w.Finalizer.Role)
+	}
+	if w.Finalizer != nil && len(w.Finalizer.Playbook) > 0 {
+		w.Finalizer.Playbook = getFullPath(w.Finalizer.Playbook)
+	}
+}
+
+// getFullPath will return a valid full path for the role and/or playbook
+func getFullPath(path string) string {
+	if isPathRelative(path) {
+		if k8sutil.IsRunModeLocal() {
+			return addCurrentDirPath(path)
+		}
```

Comment:
We should not have logic in our `watches.yaml` loader that depends on whether we are running locally or not. Instead, we should always use the path relative to the current directory and then make sure that when we're running in a cluster that our current working directory contains the files we expect, based on their relative paths.